### ISSUE TITLE
Parsing raw transactions is complicated in 2.x

### ISSUE THREAD

**Issue by alcuadrado** [ethereumjs/ethereumjs-tx#169]
I'm upgrading this library in [Buidler](https://github.com/nomiclabs/buidler/), and found that parsing raw (i.e. RLP encoded) transactions can be challenging.

If the transaction is signed using EIP155, you need to know its `chainId` to call the `Transaction` constructor. This requirement may not be realistic, as you have to parse the tx to get its `chainId`, and you parse it by calling that constructor.

Once you get the chainId, two things can happen:

1. Somewhat easy case: That chain is supported by `ethereumjs-common` and you can pass it in the `TransactionOptions` object. But how do you know if it is supported? I'm using this hack, that feels really fragile:

```ts
  const chains = require("ethereumjs-common/dist/chains");
  if (chains.chains.names[chainId] !== undefined) {
     transaction = new Transaction(tx, { chain: chainId });
  }
```

2. Harder case: the chain is not supported by `ethereumjs-common`. What should you do in that case? I'm doing this:

```ts
  } else {
    const { default: Common } = await import("ethereumjs-common");

    const common = Common.forCustomChain(
      "mainnet",
      {
        chainId,
        networkId: chainId
      },
      "petersburg"
    );

    transaction = new Transaction(tx, { common });
  }
```

I think we should improve this situation by:

1. Adding a static method to `Common.isStandardChain(chain: number|string): boolean`.

2. Add a static factory to `Transaction`. Something like `Transaction.parse(rawTx:Buffer, ...)`.  I'm not completely sure how this method should work, but I think it should be able to parse txs from standard chains without extra arguments. If a tx isn't from a standard chain, it should provide more meaningful errors.

What do you think?

**Comment by holgerd77** [comment]
Oh, that really looks a bit ugly. 🤨

With `isStandardChain` you mean a supported chain? Would directly using `isSupportedChain` be more straight-forward? I find this terminology "standard" a bit blurry (I first thought of: officially EF initiated chains or something). Otherwise I find the addition from 1. useful.

I would also cautiously think that the static factory way suggested in 2. also makes somewhat sense.

**Comment by alcuadrado** [comment]
> With `isStandardChain` you mean a supported chain?

Yes, supported chain sounds much better. I didn't like the `standard` term, thanks! 

I'm also thinking that we could instruct other chains to release their version of `-common`, so people could use them just with an import:

```js
const {Transaction} = require("ethereumjs-common")
const common = require("rsk-common")

new Transaction(..., { common })
```

This can be promoted by adding documentation and a boilerplate/template repo doing it. We are using [the boilerplate strategy for Buidler](https://github.com/nomiclabs/buidler-ts-plugin-boilerplate) and have received very positive feedback and results.
